connector/tests/common.py

```diff
@@ -1,47 +1,63 @@
+    load stub components::
 
-      'odoo.addons.magentoerpconnect.stock_picking.export_picking_done'
 
-    This is a context manager, all the calls made to the job function in
-    job_path inside the context manager will be executed synchronously.
+        from odoo.addons.connector_magento.models.product.importer import (
+            CatalogImageImporter
+        )
 
-    .. note:: It uses :meth:`mock.patch` so it has the same pitfall
-              regarding the python path.  If the mock seems to have no
-              effect, read `Where to patch
-              <http://www.voidspace.org.uk/python/mock/patch.html#where-to-patch>`_
-              in the mock documentation.
+        class TestImportProductImage(MagentoSyncTestCase):
+
+            def test_import_images(self):
+
+                # create a Stub component replacing the normal adapter
+                # (it has the same '_usage')
+                class StubProductAdapter(Component):
+
+                    _name = 'stub.product.adapter'
+                    _collection = 'magento.backend'
+                    _usage = 'backend.adapter'
+                    _apply_on = 'magento.product.product'
+
+                    def get_images(self, id, storeview_id=None):
+                        return [...]
+
+                # build the Stub and the component we want to test
+                self._build_components(StubProductAdapter,
+                                       CatalogImageImporter)
+
+                work = WorkContext(model_name='magento.product.product',
+                                   collection=self.backend,
+                                   components_registry=self.comp_registry)
+                image_importer = work.component_by_name(
+                    'magento.product.image.importer'
+                )
+                # the image importer will lookup the 'backend.adapter'
+                # component, which will our Stub, and call get_images()
+                image_importer.run(111)
+                # ... asserts below
 
     """
-    job_module, job_name = job_path.rsplit('.', 1)
-    module = importlib.import_module(job_module)
-    job_func = getattr(module, job_name, None)
-    assert job_func, "The function %s must exist in %s" % (job_name,
-                                                           job_module)
-
-    def clean_args_for_func(*args, **kwargs):
-        # remove the special args reserved to '.delay()'
-        kwargs.pop('priority', None)
-        kwargs.pop('eta', None)
-        kwargs.pop('model_name', None)
-        kwargs.pop('max_retries', None)
-        kwargs.pop('description', None)
-        job_func(*args, **kwargs)
-
-    with mock.patch(job_path) as patched_job:
-        # call the function directly instead of '.delay()'
-        patched_job.delay.side_effect = clean_args_for_func
-        yield patched_job
+
+    def setUp(self):
+        super(ConnectorTransactionCase, self).setUp()
```

Comment:
@guewen `setUpClass` is more appropriate to avoid a reload of the registry for each test.
IMO this changes breaks the modularity. If you write a test for your components you need to know all the modules you depend on that provide components to manually load their components.
You can safely rely on the `_register_hook` method of the `component.builder`model to load all the components of the modules already installed since the method will be called at the end of the registry loading process AND the method clear/clean the registry at each call. Therefore you must only call the `load_components` method for the addon you are testing. Do I miss something?